akka-remote/src/test/scala/akka/remote/classic/transport/AkkaProtocolSpec.scala

```diff
@@ -395,8 +389,6 @@ class AkkaProtocolSpec extends AkkaSpec("""akka.actor.provider = remote """) wit
           h.remoteAddress should ===(remoteAkkaAddress)
           h.localAddress should ===(localAkkaAddress)
           h
-
-        case _ => fail()
```

Comment:
What's going on here? The `fail()` is no longer necessary because the `MatchError` will fail the test anyway?

(I'm curious) How did the `case _ => fail()` cause scala 3 to complain?

Reply:
`val statusPromise: Promise[AssociationHandle]`

So it can only be a `AssociationHandle`. Scala 3 probably complained because `case _` will never happen. 

Which bring us to the question, why are we pattern matching? 😄

Reply:
> Scala 3 probably complained because `case _` will never happen.

Yes

> why are we pattern matching?

Hum, I agree it looks like it'd be good to remove the pattern-match entirely... not sure I want to grow this PR with that though :)